Add React keys in subscribe-channels (issue #8729)

jhonnyV-V opened the pull request:
## What does this PR change?

Add a key property in multiple elements, this fixes #8729  

## GUI diff

No difference.

## Documentation
- No documentation needed

## Test coverage
- No tests: already covered

## Links

Issue(s): #8729

## Changelogs
If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

Etheryte commented:
Thanks a lot for stopping by and taking the time to fix this, that's very kind of you. I have one small nitpick about the type annotation, but outside of that this looks good. Could you please run `yarn lint` to fix the premerge checks?

Etheryte commented on web/html/src/manager/systems/ssm/ssm-subscribe-channels.tsx:
```diff
@@ -406,7 +406,7 @@ class ChildChannelPage extends React.Component<ChildChannelProps, ChildChannelSt
 
   render() {
-    const rows = this.props.childChannels.map((allowed) => {
+    const rows = this.props.childChannels.map((allowed: SsmAllowedChildChannelsDto) => {
```
Here and below, this manual annotation is not really necessary since the type is already inferred from the prop type.

jhonnyV-V commented:
> Thanks a lot for stopping by and taking the time to fix this, that's very kind of you. I have one small nitpick about the type annotation, but outside of that this looks good. 👍
> 
> Could you please run `yarn lint` to fix the premerge checks?

sure, I just did the linting and removed the type annotation, I squashed the commits and force pushed to keep the history clean

Etheryte reviewed (approved):
Thanks, much appreciated. That looks good, approved.

Fyi you will not be able to merge this PR yourself, I will merge it once our release cycle has branches open again.

deneb-alpha commented:
failures on tests are unrelated. checked with QE  team before merging
